### PR TITLE
[WIP][DO NOT REVIEW] Refactor manual_seed Python API to be device-agnostic

### PR DESCRIPTION
**Pull request by shink**
add device arg to `torch.manual_seed()`

### CODE REVIEW

File: torch/random.py [hunk 1]
```diff
@@ -29,17 +30,25 @@ def get_rng_state() -> torch.Tensor:
...
+
+    # Set the seed for the current device of a given device type
+    if device is not None:
+        device_type = device.type if isinstance(device, torch.device) else device
+        _manual_seed_for_device(seed, device_type)
+        return None
+
     import torch.cuda
 
     if not torch.cuda._is_in_bad_fork():
```
Comment (shink) [inline torch/random.py:54]: Take CUDA as an example:
- If set device="cuda", `torch.cuda.manual_seed()` will be called, the seed will be set on the current GPU
- if not, `torch.cuda.manual_seed_all` will be called, the seed will be set on all GPUs
